syntax = "proto3";

package battleshipservice;

// BoardService
service BoardService {
  rpc FireMissile (FireMissileRequest) returns (FireMissileResponse) {}
  rpc StreamPlacedShips (ShowPlacedShipsRequest) returns (stream ShowPlacedShipsResponse) {}
}

message ShowPlacedShipsRequest{}

message ShowPlacedShipsResponse{
  message StreamEvent {
    int32 col = 1;
    int32 row = 2;
  }

  repeated StreamEvent events = 1;
}

message FireMissileRequest{
  int32 col = 1;
  int32 row = 2;
}

message FireMissileResponse{
  bool status = 1;
}

// Game Service

// service UnaryService {
//   rpc BattleShip (BattleShipRequest) returns (BattleShipResponse) {}
// }

// service ServerStreamingService {
//   rpc BattleShip (BattleShipRequest) returns (stream BattleShipStreamingResponse) {}
// }

// message BattleShipRequest {
//   string message = 1;
// }

// message BattleShipResponse {
//   string message = 1;
// }

// message BattleShipStreamingResponse {
//   message BattleShipStreamEvent {
//     string message = 1;
//   }

//   repeated BattleShipStreamEvent events = 1;
// }